### PR TITLE
Test gen certs command

### PR DESCRIPTION
**Pull request by MichalMendrala**

### CODE REVIEW

**Review by KonradTajstra — CHANGES_REQUESTED** [review]
As far as I can see this PR focuses on testing certificates generation, however issue 264 is about generating mocks for unit tests (in gmock) - please change the title

**Review by KonradTajstra — DISMISSED** [review]
Please squash commits when merging

File: tests/ut/ut_scargo_gen.py [hunk 1]
```diff
@@ -213,19 +222,378 @@ def test_gen_mock_for_unexpected_extension_file(
     assert not os.path.exists(f"tests/mocks/{file}")
 
 
-@pytest.mark.skip("Not implemented yet")
+@pytest.mark.parametrize(
+    "test_project_data",
+    [
+        {"proj_path": "coppy_test_project", "src_files_dir": "src"},
+        {"proj_path": "coppy_test_project_esp32", "src_files_dir": "main"},
+        {"proj_path": "coppy_test_project_stm32", "src_files_dir": "src"},
```
Comment (BartBart-pl) [inline tests/ut/ut_scargo_gen.py:230]: coppy

File: tests/ut/ut_scargo_gen.py [hunk 2]
```diff
@@ -213,19 +222,378 @@ def test_gen_mock_for_unexpected_extension_file(
...
+        assert os.path.exists(f"{CERTS_DIR}/newcerts/02.pem")
+
+        assert os.path.exists(f"{CERTS_DIR}/private")
+        assert os.path.exists(
+            f"{CERTS_DIR}/private/azure-iot-test-only.intermediate.key.pem"
+        )
+        assert os.path.exists(
+            f"{CERTS_DIR}/private/azure-iot-test-only.root.ca.key.pem"
+        )
+        assert os.path.exists(f"{CERTS_DIR}/private/iot-device.key.pem")
```
Comment (BartBart-pl) [inline tests/ut/ut_scargo_gen.py:333]: Maybe assume instead of assert to check all conditions?

**Review by BartBart-pl — APPROVED** [review]

**Review by KonradTajstra — APPROVED** [review]

**Pull request merged**